### PR TITLE
Added CSES - Counting Reorders

### PR DESCRIPTION
**Pull request by dustin-miao**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: content/6_Advanced/FFT.problems.json [hunk 1]
```diff
@@ -77,6 +77,18 @@
...
+    },
+    {
+      "uniqueId": "cses-2421",
+      "name": "Counting Reorders",
+      "url": "https://cses.fi/problemset/task/2421",
+      "source": "CSES",
+      "difficulty": "Very Hard",
+      "isStarred": false,
+      "tags": ["Knapsack", "Generating Functions", "FFT", "DP"],
+      "solutionMetadata": {
```
Comment (SansPapyrus683) [inline content/6_Advanced/FFT.problems.json:89]: is this needs fft it should not be in gold what

**Comment by SansPapyrus683** [comment]
why is there so much template code 💀

**Comment by envyaims** [comment]
mfw fft

**Comment by dustin-miao** [comment]
> why is there so much template code 💀

It's just for the modulo int, which I would argue using makes the code clearer.

**Comment by SansPapyrus683** [comment]
> mfw fft

apparently fft is just multiplying polynomials fast

> It's just for the modulo int, which I would argue using makes the code clearer.

then codesnip it

**Comment by dustin-miao** [comment]
I did?

File: solutions/gold/cses-2421.mdx [hunk 2]
```diff
@@ -0,0 +1,1403 @@
...
+$$
+and let $p(x)=\prod_c p_c(x)$. Then, we have that $f_{m,k}$ is equal to the coefficient of $x^k$ in $p(x)$. Each polynomial multiplication can be done efficiently using FFT. We can multiply the polynomials using a divide and conquer (as opposed to sequentially) to ensure that whenever a polynomial multiplication is performed, the two terms are roughly the same size.
+
+<Info title="Alternate Derivation">
+It's possible to derive this solution directly using generating functions, as described in [here](https://math.stackexchange.com/questions/4572019/arrangements-of-a-a-b-b-b-c-c-c-c-in-which-no-two-consecutive-letters-are-the) and [here](https://math.stackexchange.com/questions/129451/find-the-number-of-arrangements-of-k-mbox-1s-k-mbox-2s-cdots/129802#129802).
+</Info>
+
+### Implementation
+
+**Time Complexity**: $\mathcal{O}(n\log n)$
```
Comment (dongliu0426) [inline solutions/gold/cses-2421.mdx:685]: it's log^2 right

Comment (dustin-miao) [reply solutions/gold/cses-2421.mdx:685]: The lengths of the polynomials add up to $n$, and you only do it up to 26 times.

**Comment by SansPapyrus683** [comment]
> I did?

lmao i'm bad

**Comment by SansPapyrus683** [comment]
image

mfw
are all 500 lines necessary?

**Comment by dustin-miao** [comment]
Yes, it's directly copied from [here](https://github.com/atcoder/ac-library/blob/master/atcoder/modint.hpp) with the dependencies included.

**Comment by SansPapyrus683** [comment]
well get someone smarter like envy to approve and i'll think about what to do from then

**Comment by jessechoe10** [comment]
Ok, I think this is too much template, so I'm gonna close this PR until you remove all the templates and make the code more readable.

**Comment by dustin-miao** [comment]
Did you even look at the code in the preview or did you just look at github

**Pull request closed without merging**